measure.rb

```diff
@@ -3922,73 +3914,46 @@ def self.add_component_loads_output(runner, model)
+
     [:htg, :clg].each do |mode|
       if mode == :htg
         sign = ""
       else
         sign = "-"
       end
-      program.addLine("Set #{mode}_mode = 0")
-      program.addLine("If #{liv_load_sensors[mode].name} > 0")
-      program.addLine("  Set #{mode}_mode = 1")
-      program.addLine("EndIf")
-      surfaces_sensors.keys.each do |k|
-        program.addLine("Set #{mode}_#{k.to_s} = #{sign}hr_#{k.to_s} * #{mode}_mode")
-      end
-      nonsurf_names.each do |nonsurf_name|
-        program.addLine("Set #{mode}_#{nonsurf_name} = #{sign}hr_#{nonsurf_name} * #{mode}_mode")
-      end
-
-      # If setpoint change or partial load hour, ratio the loads to equal the total for this hour.
-      program.addLine("Set #{mode}_ratio = 0")
-      program.addLine("If (#{setpoint_sensors[mode].name} <> #{prev_hr_setpoint_vars[mode].name}) && (#{mode}_mode > 0)")
-      program.addLine("  Set #{mode}_ratio = 1")
-      program.addLine("ElseIf (#{liv_load_sensors[mode].name} * #{prev_hr_load_vars[mode].name} == 0) && (#{mode}_mode > 0)")
-      program.addLine("  Set #{mode}_ratio = 1")
-      program.addLine("EndIf")
-      program.addLine("If #{mode}_ratio > 0")
-      program.addLine("  Set #{mode}_sum = 0")
-      surfaces_sensors.keys.each do |k|
-        program.addLine("  Set #{mode}_sum = #{mode}_sum + #{mode}_#{k.to_s}")
-      end
-      nonsurf_names.each do |nonsurf_name|
-        program.addLine("  Set #{mode}_sum = #{mode}_sum + #{mode}_#{nonsurf_name}")
-      end
-      program.addLine("  If #{mode}_sum <> 0")
-      program.addLine("    Set #{mode}_load_ratio = #{liv_load_sensors[mode].name} / #{mode}_sum")
-      program.addLine("Else")
-      program.addLine("    Set #{mode}_load_ratio = 1")
-      program.addLine("EndIf")
       surfaces_sensors.keys.each do |k|
-        program.addLine("  Set #{mode}_#{k.to_s} = #{mode}_#{k.to_s} * #{mode}_load_ratio")
+        program.addLine("Set loads_#{mode}_#{k.to_s} = #{sign}hr_#{k.to_s} * #{mode}_mode")
       end
       nonsurf_names.each do |nonsurf_name|
-        program.addLine("  Set #{mode}_#{nonsurf_name} = #{mode}_#{nonsurf_name} * #{mode}_load_ratio")
+        program.addLine("Set loads_#{mode}_#{nonsurf_name} = #{sign}hr_#{nonsurf_name} * #{mode}_mode")
       end
-      program.addLine("EndIf")
-
-      program.addLine("Set #{prev_hr_load_vars[mode].name} = #{liv_load_sensors[mode].name}")
-      program.addLine("Set #{prev_hr_setpoint_vars[mode].name} = #{setpoint_sensors[mode].name}")
     end
 
+    # EMS program: Total loads
+    program.addLine("Set loads_htg_tot = 0")
+    program.addLine("Set loads_clg_tot = 0")
+    program.addLine("If #{liv_load_sensors[:htg].name} > 0")
+    program.addLine("  Set loads_htg_tot = #{tot_load_sensors[:htg].name} - #{tot_load_sensors[:clg].name}")
+    program.addLine("ElseIf #{liv_load_sensors[:clg].name} > 0")
+    program.addLine("  Set loads_clg_tot = #{tot_load_sensors[:clg].name} - #{tot_load_sensors[:htg].name}")
```

Comment:
Is it possible to have non-zero Heating:EnergyTransfer and non-zero Cooling:EnergyTransfer simultaneously in EMS (operates in time step resolution)?

Reply:
Yes, there can be non-zero Heating:EnergyTransfer in one zone and non-zero Cooling:EnergyTransfer in another zone. For example, this can happen if the ducts are in the attic and the heating system is running, but the attic zone is actually warmer than the living zone (heating setpoint). In this case, Heating:EnergyTransfer:Zone:Living and Cooling:EnergyTransfer:Zone:PlenumZone are both non-zero. 

After making the above code change, a couple test files show much better agreement between total loads and sum of component loads.

Reply:
Thanks for clarification. I thought E+ might also have offset cooling and heating between zones, like what it did of summing up all pieces of hvac equipment no matter if it's cooling only or heating only😅. Good to see better agreement!